Fix section merge alignment error message

martendo opened the pull request:
I ran into this error a little while ago and noticed it said "8-byte alignment" when I had used `ALIGN[8]`.

Rangi42 commented:
Nice observation! A test case would be good for this, to demonstrate the effect and make sure it doesn't go wrong again.

(To make one, you'd create a small file `test/asm/incompatible-alignment.asm` which would trigger this error message, then run `rgbasm -Wall incompatible-alignment.asm >incompatible-alignment.out 2>incompatible-alignment,err` to record the expected result.)

martendo commented:
I didn't realize earlier, but there are a couple test cases already in `test/link/` that had the wrong error message, so I updated those. I also added a new test like you suggested, `test/asm/incompatible-alignment.asm`.

Rangi42 commented:
Hm. The test case fails in macOS:

> ```
>  ERROR: incompatible-alignment.asm(8):
> -    Section already declared with incompatible 256-byte alignment (offset 0)
> +    Section already declared with incompatible 0-byte alignment (offset 0)
> ```

`PRIu8` is for printing 8-bit values, and `ALIGN[8]` causes `1U << 8 == 256` which overflows to 0. (Odd that it doesn't fail on everything.) Anyway, change `PRIu8` to plain `%d`, and I think `1U` can just be `1`.

ISSOtm commented on src/asm/section.c:
```diff
@@ -160,9 +160,9 @@ static unsigned int mergeSectUnion(struct Section *sect, enum SectionType type,
 		} else if ((alignOffset & mask(sect->align))
 			   != (sect->alignOfs & mask(alignment))) {
-			fail("Section already declared with incompatible %" PRIu8
+			fail("Section already declared with incompatible %d"
```
```suggestion
			fail("Section already declared with incompatible %u"
```

ISSOtm commented on src/asm/section.c:
```diff
@@ -160,9 +160,9 @@ static unsigned int mergeSectUnion(struct Section *sect, enum SectionType type,
+			fail("Section already declared with incompatible %d"
 			     "-byte alignment (offset %" PRIu16 ")\n",
-			     sect->align, sect->alignOfs);
+			     1 << sect->align, sect->alignOfs);
```
```suggestion
			     1u << sect->align, sect->alignOfs);
```

ISSOtm commented on src/asm/section.c:
```diff
@@ -213,9 +213,9 @@ static unsigned int mergeFragments(struct Section *sect, enum SectionType type,
 		/* Check if alignment offsets are compatible */
 		} else if ((curOfs & mask(sect->align)) != (sect->alignOfs & mask(alignment))) {
-			fail("Section already declared with incompatible %" PRIu8
+			fail("Section already declared with incompatible %d"
```
```suggestion
			fail("Section already declared with incompatible %u"
```

ISSOtm commented on src/asm/section.c:
```diff
@@ -213,9 +213,9 @@ static unsigned int mergeFragments(struct Section *sect, enum SectionType type,
+			fail("Section already declared with incompatible %d"
 			     "-byte alignment (offset %" PRIu16 ")\n",
-			     sect->align, sect->alignOfs);
+			     1 << sect->align, sect->alignOfs);
```
```suggestion
			     1u << sect->align, sect->alignOfs);
```
